src/JsonApi/Serializer/CollectionNormalizer.php

```diff
@@ -0,0 +1,207 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\JsonApi\Serializer;
+
+use ApiPlatform\Core\Api\ResourceClassResolverInterface;
+use ApiPlatform\Core\DataProvider\PaginatorInterface;
+use ApiPlatform\Core\Exception\RuntimeException;
+use ApiPlatform\Core\Metadata\Property\Factory\PropertyMetadataFactoryInterface;
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+use ApiPlatform\Core\Serializer\ContextTrait;
+use ApiPlatform\Core\Util\IriHelper;
+use Symfony\Component\Serializer\Normalizer\NormalizerAwareInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerAwareTrait;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+/**
+ * Normalizes collections in the JSON API format.
+ *
+ * @author Kevin Dunglas <dunglas@gmail.com>
+ * @author Hamza Amrouche <hamza@les-tilleuls.coop>
+ */
+final class CollectionNormalizer implements NormalizerInterface, NormalizerAwareInterface
+{
+    use ContextTrait;
+    use NormalizerAwareTrait;
+
+    const FORMAT = 'jsonapi';
+
+    private $resourceClassResolver;
+    private $pageParameterName;
+    private $resourceMetadataFactory;
+    private $propertyMetadataFactory;
+
+    public function __construct(
```

Comment:
As we follow Symfony CS (http://symfony.com/doc/current/contributing/code/standards.html), we always inline parameters. (We basically inline most constructs).